sahi/predict.py

```diff
@@ -460,14 +472,97 @@ def predict(
     # iterate over source images
     durations_in_seconds["prediction"] = 0
     durations_in_seconds["slice"] = 0
-    for ind, image_path in enumerate(tqdm(image_path_list, "Performing inference on images")):
+
+    # Input video
+    if export_video:
+        if video_path is not None:
+            #get video name with extension
+            image_base = os.path.basename(video_path)
+        else:
+            print("\nVideo path is none. Enter (--video_path 'path' --export_video) in terminal!\n")
+        # get video from video path
+        cap = cv2.VideoCapture(video_path)
+        # get video properties
+        fps = cap.get(cv2.CAP_PROP_FPS)
+        w = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
+        h = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
+        size = (w, h)
+
+        fourcc = cv2.VideoWriter_fourcc(*'mp4v')
+        out = cv2.VideoWriter(str(save_dir) + '/{}'.format(image_base), fourcc, fps, size)
+
+        if view_image:
+            cv2.namedWindow('Prediction of {}'.format(str(image_base)), cv2.WINDOW_AUTOSIZE)
+        
+        if forward_backward:
+            
+            def get_video(cap):
+                while cap.isOpened:
+                    k = cv2.waitKey(30)
+
+                    frame_num = cap.get(cv2.CAP_PROP_POS_FRAMES)
+
+                    if k == 27:
+                        print('\nQuitting...\n') # Exit the prediction, Key = Esc
```

Comment:
We can remove those `"\n"` (we can refactor those in the whole change), and optionally replace with something else, for example

```python
===== Closing =====
```

Reply:
When we remove those "\n",  text is written same line. But we can remove at end of text.